cmd/boots/flag_test.go

```diff
@@ -0,0 +1,117 @@
+package main
+
+import (
+	"flag"
+	"fmt"
+	"testing"
+	"time"
+
+	"github.com/google/go-cmp/cmp"
+)
+
+func TestParser(t *testing.T) {
+	want := config{
+		syslog: syslogConfig{
+			enabled:  true,
+			bindAddr: "192.168.2.4:514",
+		},
+		tftp: tftp{
+			enabled:  true,
+			timeout:  5 * time.Second,
+			bindAddr: "192.168.2.4:69",
+		},
+		ipxeHTTPBinary: ipxeHTTPBinary{
+			enabled: true,
+		},
+		ipxeHTTPScript: ipxeHTTPScript{
+			enabled:  true,
+			bindAddr: "192.168.2.4:80",
+		},
+		dhcp: dhcpConfig{
+			enabled:           true,
+			bindAddr:          "0.0.0.0:67",
+			ipForPacket:       "192.168.2.4",
+			syslogIP:          "192.168.2.4",
+			tftpIP:            "192.168.2.4:69",
+			httpIpxeBinaryIP:  "http://192.168.2.4:8080/ipxe/",
+			httpIpxeScriptURL: "http://192.168.2.4/auto.ipxe",
+		},
+		logLevel: "info",
+		backends: dhcpBackends{
+			file:       File{},
+			kubernetes: Kube{Enabled: true},
+		},
+	}
+	got := config{}
+	fs := flag.NewFlagSet(name, flag.ContinueOnError)
+	args := []string{
+		"-log-level", "info",
+		"-syslog-addr", "192.168.2.4:514",
+		"-tftp-addr", "192.168.2.4:69",
+		"-http-addr", "192.168.2.4:80",
+		"-dhcp-ip-for-packet", "192.168.2.4",
+		"-dhcp-syslog-ip", "192.168.2.4",
+		"-dhcp-tftp-ip", "192.168.2.4:69",
+		"-dhcp-http-ipxe-binary-ip", "http://192.168.2.4:8080/ipxe/",
+		"-dhcp-http-ipxe-script-url", "http://192.168.2.4/auto.ipxe",
+	}
+	cli := newCLI(&got, fs)
+	cli.Parse(args)
+	opts := cmp.Options{
+		cmp.AllowUnexported(config{}),
+		cmp.AllowUnexported(syslogConfig{}),
+		cmp.AllowUnexported(tftp{}),
+		cmp.AllowUnexported(ipxeHTTPBinary{}),
+		cmp.AllowUnexported(ipxeHTTPScript{}),
+		cmp.AllowUnexported(dhcpConfig{}),
+		cmp.AllowUnexported(dhcpBackends{}),
+	}
+	if diff := cmp.Diff(want, got, opts); diff != "" {
+		t.Fatal(diff)
+	}
+}
+
+func TestCustomUsageFunc(t *testing.T) {
+	defaultIP := detectPublicIPv4("")
+	want := fmt.Sprintf(`USAGE
+  Run Boots server for provisioning
+
+FLAGS
```

Comment:
There are a number of acronym inconsistencies in the help. `ip` vs `IP`, `http` vs `HTTP`.

Reply:
fixed